python/tank/platform/validation.py

```diff
@@ -285,9 +285,15 @@ def __validate_schema_value(self, settings_key, schema):
         data_type = schema.get("type")
         self.__validate_schema_type(settings_key, data_type)
 
-        if "default_value" in schema:
+        # Get a list of all keys that start with the default value key string.
+        # Some types, like hooks, allow for engine specific default values such
+        # as "default_value_tk-maya". Validate each of these key's values
+        # against the setting's data type.
+        default_value_keys = [k for k in schema if k.startswith("default_value")]
```

Comment:
Shouldn't this test be setting type specific?  Since this would only apply to hooks?  There would be an edge case if you had a string that started with "default_value", wouldn't you?